src/os/bluestore/BlueStore.cc

```diff
@@ -5782,12 +5782,12 @@ int64_t BlueStore::_get_bluefs_size_delta(uint64_t bluefs_free, uint64_t bluefs_
   uint64_t gift = 0;
   uint64_t reclaim = 0;
   if (bluefs_ratio < cct->_conf->bluestore_bluefs_min_ratio) {
-    gift = cct->_conf->bluestore_bluefs_gift_ratio * total_free;
+    gift = cct->_conf->bluestore_bluefs_gift_ratio * my_free;
```

Comment:
these config options are meant to be a fraction of the total freespace, not the part claimed by bluefs or bluestore.  apply a ceiling?

Reply:
e.g., bluefs_free 100G, bluestore_free 2G, total_free = 102G, then
gift = 0.02 * 102G =  2.04G, which is greater than bluestore_free 2G

In general this should not happen, but recently I played with some unusual  bluestore_bluefs_min_ratio/bluestore_bluefs_max_ratio values to study the space balancing behavior and hit a assert caused by the above space overflow..

Reply:
In that case, we should limit the gift size.  but in the general case, i think we should keep the current meaning of the config option, which is .02 of total freespace?